python/tests/io_tests/files/test_find.py

```diff
@@ -0,0 +1,117 @@
+            actual_result = is_aac_file(test_yaml.name)
+
+        self.assertFalse(actual_result)
+
+    def test_is_aac_file_with_invalid_file_extension(self):
+        test_file_content = get_aac_spec_as_yaml()
+        with temporary_test_file(test_file_content, suffix=".notYamlOrAac") as test_yaml:
+            actual_result = is_aac_file(test_yaml.name)
+
+        self.assertFalse(actual_result)
+
+    def test_find_aac_files(self):
+
+        with TemporaryDirectory() as l1_temp_dir:
+            l1_aac_file_aac = NamedTemporaryFile(dir=l1_temp_dir, suffix=AAC_FILE_EXTENSION, mode="w")
+            _write_content_to_temp_file(l1_aac_file_aac, get_aac_spec_as_yaml())
+
+            l1_invalid_file_aac = NamedTemporaryFile(dir=l1_temp_dir, suffix=AAC_FILE_EXTENSION, mode="w")
+            _write_content_to_temp_file(l1_invalid_file_aac, VALID_NON_AAC_YAML_CONTENT)
+
+            with TemporaryDirectory(dir=l1_temp_dir) as l2_temp_dir:
+                l2_aac_file_yaml = NamedTemporaryFile(dir=l2_temp_dir, suffix=YAML_FILE_EXTENSION, mode="w")
+                _write_content_to_temp_file(l2_aac_file_yaml, get_aac_spec_as_yaml())
+
+                l2_invalid_file_yaml = NamedTemporaryFile(dir=l2_temp_dir, suffix=YAML_FILE_EXTENSION, mode="w")
+                _write_content_to_temp_file(l2_invalid_file_yaml, VALID_NON_AAC_YAML_CONTENT)
+
+                with TemporaryDirectory(dir=l2_temp_dir) as l3_temp_dir:
+                    l3_aac_file_aac = NamedTemporaryFile(dir=l3_temp_dir, suffix=AAC_FILE_EXTENSION, mode="w")
+                    _write_content_to_temp_file(l3_aac_file_aac, get_aac_spec_as_yaml())
+
+                    l3_invalid_file_aac = NamedTemporaryFile(dir=l3_temp_dir, suffix=AAC_FILE_EXTENSION, mode="w")
+                    _write_content_to_temp_file(l3_invalid_file_aac, VALID_NON_AAC_YAML_CONTENT)
+
+                    expected_result = [l1_aac_file_aac.name, l2_aac_file_yaml.name, l3_aac_file_aac.name]
+                    actual_result = find_aac_files(l1_temp_dir)
+
+                    # Have to manually close the temp files here, otherwise their temp directory context managers will cause test errors.
+                    l1_aac_file_aac.close()
+                    l1_invalid_file_aac.close()
+
+                    l2_aac_file_yaml.close()
+                    l2_invalid_file_yaml.close()
+
+                    l3_aac_file_aac.close()
+                    l3_invalid_file_aac.close()
+
+        actual_result_file_paths = [file.uri for file in actual_result]
+        self.assertListEqual(expected_result, actual_result_file_paths)
+        self.assertNotIn(l1_invalid_file_aac.name, actual_result_file_paths)
+        self.assertNotIn(l2_invalid_file_yaml.name, actual_result_file_paths)
+        self.assertNotIn(l3_invalid_file_aac.name, actual_result_file_paths)
+
+
+def _write_content_to_temp_file(temp_file, content: str) -> None:
```

Comment:
Why use this instead of `temporary_test_file`? It seems like the places this is used should be able to be done using that helper.

Reply:
I think the way to get multiple CMs in the same statement is to use parentheses - at least that's how I've made it work previously.

Reply:
I updated the test with the parentheses. Not sure why it didn't work the first time I tried with parentheses. 🤷‍♂️